### PR TITLE
builder: add email address with hidden field

### PR DESCRIPTION
**Pull request by harunurhan**
* Adds or updates (if hidden is not None) with `hidden` field to
indicate if email is public or not.

* NSPIR-2157

### CODE REVIEW

File: inspire_schemas/builders/authors.py [hunk 1]
```diff
@@ -110,18 +110,27 @@ def add_native_name(self, name):
...
+        found_email = next(
+            (existing_email for existing_email in existing_emails if existing_email.get('value') == email),
+            None)
+        if found_email is None:
+            new_email = {'value': email}
+            if hidden is not None:
+                new_email['hidden'] = hidden
+            self._append_to('email_addresses', new_email)
+        elif hidden is not None:
+            found_email['hidden'] = hidden
```
Comment (drjova) [inline inspire_schemas/builders/authors.py]: ```self._append_to('email_addresses', new_email)```?

File: inspire_schemas/builders/authors.py [hunk 2]
```diff
@@ -110,18 +110,27 @@ def add_native_name(self, name):
...
         """
-        existing_emails = get_value(self.obj, 'email_addresses.value', [])
-        if email not in existing_emails:
-            self._append_to('email_addresses', {
-                "value": email
-            })
+        existing_emails = get_value(self.obj, 'email_addresses', [])
+        found_email = next(
+            (existing_email for existing_email in existing_emails if existing_email.get('value') == email),
+            None)
```
Comment (drjova) [inline inspire_schemas/builders/authors.py]: this is confusing to read, maybe with better indentation will be easier to follow:

```python
found_email = next(
    (existing_email for existing_email in existing_emails if existing_email.get('value') == email),
    None
)
```

**Comment by drjova** [comment]
isort issues on travis

File: inspire_schemas/builders/authors.py (hunk 2 above)
Comment (harunurhan) [reply inspire_schemas/builders/authors.py]: I would prefer a regular `for` loop but it's not "pythonic" 😄

**Comment by harunurhan** [comment]
@drjova not caused by the PR :/, isort is supposed to fix this file.

**Comment by drjova** [comment]
@harunurhan commit message typo ``INSPIR-2157``

File: inspire_schemas/builders/authors.py (hunk 2 above)
Comment (drjova) [reply inspire_schemas/builders/authors.py]: you can always do it simple:

```
existing_emails = [existing_email for existing_email in existing_emails if existing_email.get('value') == email]

if existing_emails:
```

**Comment by drjova** [comment]
indeed isort should ignore this file

**Comment by drjova** [comment]
In the latest isort, released few days ago:

Fixed issue #909 - skip and skip-glob are not enforced when using settings-path.

File: inspire_schemas/builders/authors.py (hunk 2 above)
Comment (harunurhan) [reply inspire_schemas/builders/authors.py]: this won't break the loop.

**Pull request merged**